### PR TITLE
Add CrUX Dashboard launcher directly to CrUX docs

### LINKED ISSUE #5730: Add CrUX Dash Launcher functionality to the CrUX docs
The CrUX Dash docs link to the experimental CrUX Dash Launcher project I wrote to give users an easier way to access the evergreen CrUX Dash: https://developer.chrome.com/docs/crux/dashboard/#accessing-the-crux-dashboard links to https://rviscomi.github.io/crux-dash-launcher/.

The launcher is a simple text input field with a very small script to redirect the user to the appropriate CrUX Dash URL. If possible, integrate this functionality directly into the CrUX Dash docs. It'd be one less click for users and also a centralized (non-personal) way to maintain the launcher.

I can redirect the launcher app to the CrUX Dash docs and shut it down eventually.

### PR DESCRIPTION
**Pull request by tunetheweb**
Fixes #5730

Changes proposed in this pull request:

- Adds a CrUX Dashboard launcher directly to CrUX docs to save directing people to @rviscomi 's project for this.

### CODE REVIEW

File: site/en/docs/crux/dashboard/index.md [hunk 1]
```diff
@@ -41,32 +41,37 @@ The CrUX Dashboard is a [Looker Studio](https://cloud.google.com/looker-studio)
...
   #origin[aria-invalid=true] {
     outline: red dashed 2px;
   }
+  #origin-submit {
+    line-height: 2.25em;
+  }
 </style>
 
 <form id="form">
   <input id="origin" aria-label="Origin" type="text" placeholder="https://developer.chrome.com" required>
```
Comment (malchata) [inline site/en/docs/crux/dashboard/index.md:70]: This input needs a `<label>` element per accessibility guidelines.

**Review by malchata — CHANGES_REQUESTED** [review]

File: site/en/docs/crux/dashboard/index.md (hunk 1 above)
Comment (tunetheweb) [reply site/en/docs/crux/dashboard/index.md:70]: Same comment as above, answered in https://github.com/GoogleChrome/developer.chrome.com/pull/5732#discussion_r1144912418

Comment (malchata) [reply site/en/docs/crux/dashboard/index.md:70]: Ah, I see in the W3C accessibility guidelines that an `aria-label` is an acceptable alternative. I'll approve then. Sorry!

File: site/en/docs/crux/dashboard/index.md [hunk 2]
```diff
@@ -41,11 +41,68 @@ The CrUX Dashboard is a [Looker Studio](https://cloud.google.com/looker-studio)
...
+var origin = document.getElementById('origin');
+form.addEventListener('submit', function(e) {
+  try {
+    origin.removeAttribute('aria-invalid');
+    e.preventDefault();
+    var url = origin.value.trim();
+    if (!url.startsWith('http')) url = 'https://' + url;
+    url = new URL(url);
+    var encoded_origin = encodeURIComponent(url.origin);
+    var url = 'https://datastudio.google.com/reporting/bbc5698d-57bb-4969-9e07-68810b9fa348/page/keDQB?params=%7B%22origin%22:%22' + encoded_origin + '%22%7D';
```
Comment (rviscomi) [inline site/en/docs/crux/dashboard/index.md:87]:
```suggestion
    var url = 'https://lookerstudio.google.com/reporting/bbc5698d-57bb-4969-9e07-68810b9fa348/page/keDQB?params=%7B%22origin%22:%22' + encoded_origin + '%22%7D';
```

File: site/en/docs/crux/dashboard/index.md [hunk 3]
```diff
@@ -41,11 +41,68 @@ The CrUX Dashboard is a [Looker Studio](https://cloud.google.com/looker-studio)
 
 ## Accessing the CrUX Dashboard
 
-To use the existing CrUX Dashboard you need to [lauch the dashboard, passing the origin as a query param](https://datastudio.google.com/u/0/reporting/bbc5698d-57bb-4969-9e07-68810b9fa348/page/keDQB?params=%7B%22origin%22:%22wwww.example.com%22%7D). To make this easier–since the origin needs to be URL encoded—you can use [Rick Viscomi](https://twitter.com/rick_viscomi)'s [CrUX Dash Launcher](https://rviscomi.github.io/crux-dash-launcher/) which takes the URL as an input and constructs the dashboard URL.
+To use the existing CrUX Dashboard you need to lauch the dashboard, passing the origin as a query param. To make this easier–since the origin needs to be URL encoded—you can use the below launcher, which takes an origin or URL as an input and launches the appropriate dashboard URL for that origin:
```
Comment (rviscomi) [inline site/en/docs/crux/dashboard/index.md:44]: Any reason not to simplify?
```suggestion
To launch the CrUX Dashboard, enter an origin or URL:
```

File: site/en/docs/crux/dashboard/index.md [hunk 4]
```diff
@@ -41,11 +41,68 @@ The CrUX Dashboard is a [Looker Studio](https://cloud.google.com/looker-studio)
...
+  try {
+    origin.removeAttribute('aria-invalid');
+    e.preventDefault();
+    var url = origin.value.trim();
+    if (!url.startsWith('http')) url = 'https://' + url;
+    url = new URL(url);
+    var encoded_origin = encodeURIComponent(url.origin);
+    var url = 'https://datastudio.google.com/reporting/bbc5698d-57bb-4969-9e07-68810b9fa348/page/keDQB?params=%7B%22origin%22:%22' + encoded_origin + '%22%7D';
+    window.location = url;
+  } catch {
```
Comment (rviscomi) [inline site/en/docs/crux/dashboard/index.md:89]: Will this actually catch invalid inputs? If I enter `test` and click submit, it redirects me to a broken dashboard. The existing launcher used `input[type=url]` to catch these kinds of errors natively, WDYT about using that here?

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: site/en/docs/crux/dashboard/index.md (hunk 4 above)
Comment (tunetheweb) [reply site/en/docs/crux/dashboard/index.md:101]: I wanted the ability to not have to specify scheme for simplicity. So you can enter `web.dev` and it works for example. `type=url` is a bit more restrictive for things like that.

The existing launcher does catch `test` but does not catch `http://test` as that is a valid URL (though not a valid external one - `http://localhost` is an example of a one word valid url) and I'm now deliberately added a default `https://` scheme if one is not provided.

URL parsing [is apparently complex](https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url) so `new URL()` is the best way of checking this IMHO but does allow single word domains.

So I added an extra check to ensure there's at least one dot in there (and not at beginning or end) in https://github.com/GoogleChrome/developer.chrome.com/pull/5732/commits/dc91e6354be4d673055f251b5d9421e1c609e5ca

File: site/en/docs/crux/dashboard/index.md [hunk 5]
```diff
@@ -41,33 +41,98 @@ The CrUX Dashboard is a [Looker Studio](https://cloud.google.com/looker-studio)
...
+    max-width: 800px;
+  }
+  #origin {
+    width: 100%;
+    font-size: 2em;
+    line-height: 1.3em;
+    text-align: center;
+  }
+  #origin[aria-invalid=true] {
+    outline: red dashed 2px;
```
Comment (rviscomi) [inline site/en/docs/crux/dashboard/index.md:62]: outline:

image


border:
image


```suggestion
    border: red dashed 2px;
```

**Review by rviscomi — CHANGES_REQUESTED** [review]

**Review by rviscomi — APPROVED** [review]
Thanks for making this change. I really like the new interactivity and simpler DX.

**Comment by tunetheweb** [comment]
Yeah agreed. Should have done it ages ago! Didn't realise it was so simple.

**Pull request merged**